python/hidet/cuda/stream.py

```diff
@@ -191,8 +194,9 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         from hidet.ffi import runtime_api
 
         current_streams = _current_streams
-        current_streams[self.device] = self.prev_stream
-        runtime_api.set_current_stream(self.prev_stream)
+        if self.prev_stream is not None:
+            current_streams[self.device] = self.prev_stream
+            runtime_api.set_current_stream(self.prev_stream)
```

Comment:
Thanks for noticing this bug. By default, all devices are on the "default stream" (in hidet, it is `default_stream(device=device_id)`). The updated implementation will not reset to the default stream when user exits the StreamContext. 

Consider implement in this way:
```
if self.device not in current_streams:
    current_streams[self.device] = default_stream(self.device)
...
```
and keep the original version.

Reply:
thanks!